timesketch/lib/utils.py

```diff
@@ -83,6 +84,31 @@ def _scrub_special_tags(dict_obj):
             _ = dict_obj.pop(field)
 
 
+def refresh_and_validate_list_of_indices(indices, datastore):
+    """Returns a list of valid indices.
+
+    This function takes a list of indices, tries to refresh them
+    and remove indices that are not found.
+
+    Args:
+        indices (list): List of indices.
+        datastore (ElasticsearchDataStore): a data store object.
+
+    Returns:
+        list of indices that are discovered.
+    """
+    for index in indices:
+        try:
+            datastore.client.indices.refresh(index=index)
```

Comment:
How expensive is this? We are doing this on every search. Is it maybe enough to do this check in the Sketch (GET) resource?

Reply:
I measured it, it seems fairly fast, but I also checked the `if index exists` and it seems about the same speed, but then a lot less overhead with the try/except, so I swapped the implementation.